File: spec/python/spec/test_expr_io_eof_bits.py
import unittest

from testformats.expr_io_eof_bits import ExprIoEofBits

class TestExprIoEofBits(unittest.TestCase):
    def test_expr_io_eof_bits(self):
        with ExprIoEofBits.from_file('src/nav_parent_switch.bin') as r:
            r._read()

            self.assertEqual(r.foo, 5167)
            self.assertEqual(r.bar, 15)
            self.assertFalse(hasattr(r, 'baz'))
            self.assertEqual(r.align, b'')
            self.assertFalse(hasattr(r, 'qux'))

File: spec/java/src/io/kaitai/struct/spec/TestExprIoEofBits.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ExprIoEofBits;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestExprIoEofBits extends CommonSpec {
    @Test
    public void testExprIoEofBits() throws Exception {
        ExprIoEofBits r = ExprIoEofBits.fromFile(SRC_DIR + "nav_parent_switch.bin");
        r._read();

        assertIntEquals(r.foo(), 5167);
        assertIntEquals(r.bar(), 15);
        assertNull(r.baz());
        assertEquals(r.align(), new byte[] {  });
        assertNull(r.qux());
    }
}
